File: frontend2/Rugby/src/app/classes/staff.ts
import { Profile } from './profile';
import { StaffType} from './stafftype';

export class Staff {
  constructor(
  public id?: number,
  public staffNumber?: string,
  public profile?: Profile,
  public staffType?: any
  ) {
    this.id = id ? id : 0;
    this.staffNumber = staffNumber ? staffNumber : '';
    this.profile = profile ? profile : new Profile();
    this.staffType = staffType ? staffType : new StaffType();
  }
}

File: backend/src/main/java/pt/isel/rugby/model/Staff.java
package pt.isel.rugby.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name ="staff")
@Data
public class Staff implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;


    @Column(name = "staff_number")
    private String staffNumber;


    @OneToOne
    @JoinColumn(name = "profileId", referencedColumnName = "id")
    private Profile profile;

    @ManyToOne
    private StaffType staffType;

}
